api/build.gradle

```diff
@@ -316,7 +346,38 @@ sourceSets {
     compileClasspath += generated.output
     runtimeClasspath += generated.output
   }
-  integration {
+
+  //
+  // Custom SourceSets
+  //
+  // Sometimes, in order to fight the clean code war, you have to get dirty in a battle.
+  // Here we use a naming convention with __ word boundaries at start and end
+  // to avoid confusion. For example, the plugin may define a __tools__Compile configuration
+  // or a compile__tools__Java task. This "shields up" approach makes it obvious where there's
+  // a dynamically defined symbol name, so usages stick out and it's easier to identify the
+  // constituent parts. Why four underscores where two might do? Groovy is making me miss C++
```

Comment:
nit: Not that I don't like reading this, but I'd save the editorialized first person narration for the PR review and just leave the essentials in the code itself.

Reply:
Yeah, people finding their afternoon taking a detour into the build script generally aren't in the most jocular mood.

By the way, IntelliJ annotates the directories associated with the source sets in the project pane:
![Screen Shot 2020-04-16 at 9 32 21 AM](https://user-images.githubusercontent.com/53479492/79462207-31434b00-7fc5-11ea-89ea-0d60f958b6d3.png)